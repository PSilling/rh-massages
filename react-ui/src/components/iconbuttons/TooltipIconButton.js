// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Button, Tooltip } from "reactstrap";

// util imports
import Util from "../../util/Util";

/**
 * Labeless icon only button with an attached tooltip.
 */
class TooltipIconButton extends Component {
  state = { tooltipActive: false };

  tooltipTarget = Util.getTooltipTargets(1)[0];

  toggleTooltip = () => {
    this.setState(prevState => ({ tooltipActive: !prevState.tooltipActive }));
  };

  render() {
    const { disabled, icon, ...rest } = this.props;
    return (
      <span className="no-print">
        <Button {...rest} id={this.tooltipTarget} outline disabled={this.props.disabled}>
          <span className={`fas fa-${this.props.icon}`} />
        </Button>
        {this.props.tooltip !== "" && !this.props.disabled && (
          <Tooltip
            isOpen={this.state.tooltipActive}
            target={this.tooltipTarget}
            toggle={this.toggleTooltip}
            trigger="hover"
          >
            {this.props.tooltip}
          </Tooltip>
        )}
      </span>
    );
  }
}

TooltipIconButton.propTypes = {
  /** icon to be shown instead of a normal label */
  icon: PropTypes.string.isRequired,
  /** function to be called on button click */
  onClick: PropTypes.func.isRequired,
  /** whether the button should be disabled */
  disabled: PropTypes.bool,
  /** button size class */
  size: PropTypes.string,
  /** button component style */
  style: PropTypes.shape({
    border: PropTypes.string
  }),
  /** tooltip to be shown on button hover */
  tooltip: PropTypes.string
};

TooltipIconButton.defaultProps = {
  disabled: false,
  size: "sm",
  style: { border: "0px solid transparent" },
  tooltip: ""
};

export default TooltipIconButton;
