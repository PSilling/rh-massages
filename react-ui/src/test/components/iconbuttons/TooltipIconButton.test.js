// react imports
import React from "react";
import { shallow } from "enzyme";

// test imports
import { Button, Tooltip } from "reactstrap";
import TooltipIconButton from "../../../components/iconbuttons/TooltipIconButton";

test("renders content with correct props", () => {
  const testFunction = jest.fn();
  const wrapper = shallow(<TooltipIconButton icon="edit" onClick={testFunction} tooltip="tooltip" />);
  const button = wrapper.find(Button);
  const tooltip = wrapper.find(Tooltip);
  const spans = wrapper.find("span");

  expect(testFunction).not.toHaveBeenCalled();

  button.props().onClick();

  expect(testFunction).toHaveBeenCalledTimes(1);
  expect(button.props().onClick).toBe(testFunction);
  expect(button.props().style).toEqual({ border: "0px solid transparent" });
  expect(button.props().size).toEqual("sm");
  expect(button.props().disabled).toBe(false);
  expect(tooltip.props().isOpen).toBe(false);
  expect(tooltip.props().children).toEqual("tooltip");
  expect(tooltip.props().target).toEqual(button.props().id);
  expect(spans.get(1).props.className).toEqual("fas fa-edit");
  expect(wrapper).toMatchSnapshot();
});
