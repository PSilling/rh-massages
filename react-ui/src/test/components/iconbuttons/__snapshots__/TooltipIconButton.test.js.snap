// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TooltipIconButton
    disabled={false}
    icon="edit"
    onClick={[Function]}
    size="sm"
    style={
        Object {
            "border": "0px solid transparent",
          }
    }
    tooltip="tooltip"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          color="secondary"
          disabled={false}
          id="Tooltip1"
          onClick={[Function]}
          outline={true}
          size="sm"
          style={
                    Object {
                              "border": "0px solid transparent",
                            }
          }
          tag="button"
          tooltip="tooltip"
>
          <span
                    className="fas fa-edit"
          />
</Button>,
        <Tooltip
          autohide={true}
          delay={
                    Object {
                              "hide": 250,
                              "show": 0,
                            }
          }
          hideArrow={false}
          isOpen={false}
          placement="top"
          placementPrefix="bs-tooltip"
          target="Tooltip1"
          toggle={[Function]}
          trigger="hover"
>
          tooltip
</Tooltip>,
      ],
      "className": "no-print",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <span
            className="fas fa-edit"
/>,
          "color": "secondary",
          "disabled": false,
          "id": "Tooltip1",
          "onClick": [Function],
          "outline": true,
          "size": "sm",
          "style": Object {
            "border": "0px solid transparent",
          },
          "tag": "button",
          "tooltip": "tooltip",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-edit",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autohide": true,
          "children": "tooltip",
          "delay": Object {
            "hide": 250,
            "show": 0,
          },
          "hideArrow": false,
          "isOpen": false,
          "placement": "top",
          "placementPrefix": "bs-tooltip",
          "target": "Tooltip1",
          "toggle": [Function],
          "trigger": "hover",
        },
        "ref": null,
        "rendered": "tooltip",
        "type": [Function],
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            color="secondary"
            disabled={false}
            id="Tooltip1"
            onClick={[Function]}
            outline={true}
            size="sm"
            style={
                        Object {
                                    "border": "0px solid transparent",
                                  }
            }
            tag="button"
            tooltip="tooltip"
>
            <span
                        className="fas fa-edit"
            />
</Button>,
          <Tooltip
            autohide={true}
            delay={
                        Object {
                                    "hide": 250,
                                    "show": 0,
                                  }
            }
            hideArrow={false}
            isOpen={false}
            placement="top"
            placementPrefix="bs-tooltip"
            target="Tooltip1"
            toggle={[Function]}
            trigger="hover"
>
            tooltip
</Tooltip>,
        ],
        "className": "no-print",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <span
              className="fas fa-edit"
/>,
            "color": "secondary",
            "disabled": false,
            "id": "Tooltip1",
            "onClick": [Function],
            "outline": true,
            "size": "sm",
            "style": Object {
              "border": "0px solid transparent",
            },
            "tag": "button",
            "tooltip": "tooltip",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-edit",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autohide": true,
            "children": "tooltip",
            "delay": Object {
              "hide": 250,
              "show": 0,
            },
            "hideArrow": false,
            "isOpen": false,
            "placement": "top",
            "placementPrefix": "bs-tooltip",
            "target": "Tooltip1",
            "toggle": [Function],
            "trigger": "hover",
          },
          "ref": null,
          "rendered": "tooltip",
          "type": [Function],
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
